engines/napi-preview-feature/utils.ts

```diff
@@ -0,0 +1,215 @@
+    ...defaultExecaOptions,
+    env,
+  })
+}
+
+async function cleanFilesystem() {
+  fs.rmdirSync('./node_modules/prisma', { recursive: true })
+  fs.rmdirSync('./node_modules/@prisma', { recursive: true })
+  fs.rmdirSync('./node_modules/.prisma', { recursive: true })
+  if (fs.existsSync('./prisma/schema.prisma')) {
+    fs.unlinkSync('./prisma/schema.prisma')
+  }
+  if (fs.existsSync('./data.json')) {
+    fs.unlinkSync('./data.json')
+  }
+  // Also remove download cache, see https://github.com/prisma/prisma/issues/7777
+  fs.rmdirSync('./node_modules/.cache/prisma', { recursive: true })
+  fs.rmdirSync(path.join(os.homedir(), '.cache/prisma'), { recursive: true })
+}
+
+export async function install(env?: Record<string, string>) {
+  await execa('yarn', ['install', '--force'], {
+    ...defaultExecaOptions,
+    env,
+  })
+}
+
+export async function version(env?: Record<string, string>) {
+  const result = await execa('yarn', ['-s', 'prisma', '-v'], {
+    ...defaultExecaOptions,
+    stdio: 'pipe',
+    env,
+  })
+  return result.stdout
+}
+
+function snapshotDirectory(path: string, hint?: string) {
+  const files = fs.readdirSync(path)
+  const snapshotName = path + ((hint) ? ' @ ' + hint : '')
+  expect(files).toMatchSnapshot(snapshotName)
+}
+
+async function testGeneratedClient(env?: Record<string, string>) {
+  await execa.node('./test-generated-client.js', [], {
+    ...defaultExecaOptions,
+    env,
+  })
+  const data = require('./data.json')
+  // using inline snapshot here as this is identical for all tests run here
+  expect(data).toMatchInlineSnapshot(`
+    Object {
+      "delete": Object {
+        "posts": Object {
+          "count": 1,
+        },
+        "users": Object {
+          "count": 1,
+        },
+      },
+      "post.findUnique": Object {
+        "id": 1,
+        "title": "Test",
+        "userId": 1,
+      },
+      "user.create": Object {
+        "email": "test@example.com",
+        "id": 1,
+        "name": "Test",
+        "posts": Array [
+          Object {
+            "id": 1,
+            "title": "Test",
+            "userId": 1,
+          },
+        ],
+      },
+      "users.findMany": Array [],
+    }
+  `)
+}
+
+function sanitizeVersionSnapshot(str: string): string {
+  let lines = str.split('\n')
+  return lines
+    .map((line) => {
+      if (line.includes('Preview Features')) {
+        return line
+      }
+      const test = line.split(':')
+      const location = test[1].match(/\(([^)]+)\)/)
+      return `${test[0]} : placeholder ${location ? location[0] : ''}`
+    })
+    .join('\n')
+}
+
+export async function runTest(options: {
+  previewFeatures?: string[]
+  binaryTargets?: string[]
+  env?: Record<string, string>
+  env_on_deploy?: Record<string, string>
+}) {
+  // This ensures that if PRISMA_FORCE_NAPI is set for the workflow it is removed before running these tests
+  // TODO Should not be necessary any more
+  if (process.env.PRISMA_FORCE_NAPI === 'true') {
```

Comment:
```suggestion
  if (process.env.PRISMA_FORCE_NAPI) {
```
So we always remove it if set?